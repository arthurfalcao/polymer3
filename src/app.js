import { PolymerElement, html } from '@polymer/polymer';

class App extends PolymerElement {
  static get is() {
    return 'polymer3-app';
  }

  static get template() {
    return html`
      <h1>Polymer3 Playground</h1>
    `;
  }
}

customElements.define(App.is, App);
