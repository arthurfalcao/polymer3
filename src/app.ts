import { PolymerElement, html } from '@polymer/polymer'
import { customElement, property } from '@polymer/decorators'
import '@polymer/app-route/app-location'
import '@polymer/app-route/app-route'
import '@polymer/iron-pages/iron-pages'

import './pages'
import './pages/404'

@customElement('polymer3-app')
class App extends PolymerElement {
  @property({ type: Object })
  data = {}

  @property({ type: Object })
  route = {}

  @property({ type: Object })
  subroute = {}

  static get template() {
    return html`
      <h1>Polymer3 Playground</h1>

      <app-location route="{{route}}"></app-location>
      <app-route
        route="{{route}}"
        pattern="/:page"
        data="{{data}}"
        tail="{{subroute}}"
      >
      </app-route>

      <iron-pages
        selected="{{data.page}}"
        attr-for-selected="id"
        fallback-selection="404"
      >
        <p3-index></p3-index>
        <p3-404 id="404"></p3-404>
      </iron-pages>
    `
  }
}

export default App
