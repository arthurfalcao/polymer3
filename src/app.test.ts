import { render } from './utils/tests/helpers'

import './app'

describe('<polymer3-app />', () => {
  it('should render the heading', () => {
    const { container, getByRole } = render(`<polymer3-app></polymer3-app>`)

    expect(
      getByRole('heading', { name: /polymer3 playground/i })
    ).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
