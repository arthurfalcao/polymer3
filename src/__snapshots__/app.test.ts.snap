// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<polymer3-app /> should render the heading 1`] = `
<polymer3-app>
   
  <h1>
    Polymer3 Playground
  </h1>
   
</polymer3-app>
`
